Sources/Architecture/Coordinator.swift

```diff
@@ -162,7 +162,7 @@ public protocol CoordinatorDelegate: class {
 }
 
 /// Used typically on view controllers to refer to it's coordinator
-public protocol Coordinated {
+public protocol Coordinated: class where Self: UIViewController {
```

Comment:
I think we'll need a third try 😃 
![snimek obrazovky 2017-07-03 v 21 20 28](https://user-images.githubusercontent.com/1540564/27805524-7ab4d7a2-6035-11e7-87aa-b8adef193709.png)